### ISSUE TITLE
Waiting in an interrupt.

### ISSUE THREAD

**Issue by Koepel** [JankBot/teensy#1]
Could you remove the `while (Wire.available() < 2);` in "controlEvent()" in "teensy.ino" ?

It is never okay to wait in an interrupt function, and there is nothing to wait for.

I prefer to check if the number of received bytes matches the number that is expected:
```
volatile byte bot_speed = 90;
volatile dir = 90;

void controlEvent(int howMany) {
  if( howMany == 2) {
    bot_speed = Wire.read();
    dir = Wire.read();
  }
}
```

**Comment by dylanhart** [comment]
We noticed this earlier and forgot to make the change. Will definitely do.

**Comment by dylanhart** [comment]
Fixed in c5a8010e6324c7e0e30e16df36da72469e9a8d8a

**Comment by Koepel** [comment]
You might want to make bot_speed and dir "volatile" variables, because they are used in a interrupt **and** in the Arduino loop().

The keyword "volatile" tells the compiler that the variables can suddenly change and that they should always be written to and read from their memory location. Without the variables being "volatile", the compiler might optimize the Arduino loop() and keep those variables in registers.
